localstack/services/events/models.py

```diff
@@ -102,3 +104,18 @@ class InvalidEventPatternException(Exception):
     def __init__(self, reason=None, message=None) -> None:
         self.reason = reason
         self.message = message or f"Event pattern is not valid. Reason: {reason}"
+
+
+class FormattedEvent(TypedDict):
+    version: str
+    id: str
+    detail_type: Optional[str]  # key "detail-type" is automatically interpreted as detail_type
```

Comment:
@maxhoheiser This type definition allows both `detail-type` and `detail_type`, but we should restrict it to the hyphen-only variant.

Python [supports an alternative syntax](https://typing.readthedocs.io/en/latest/spec/typeddict.html#alternative-syntax) for defining TypedDicts, which is recommended in such cases:
> The functional syntax should also be used when any of the keys are not valid [identifiers](https://docs.python.org/3/reference/lexical_analysis.html#identifiers), for example because they are keywords or contain hyphens.
https://docs.python.org/3/library/typing.html#typing.TypedDict

All credits go to @dfangl for this suggestion 💯